src/pages/Order/index.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  Box,
+  Button,
+  ChakraProvider,
+  Checkbox,
+  Flex,
+  HStack,
+  Image,
+  Input,
+  Select,
+  Text,
+  Textarea,
+  VStack,
+} from '@chakra-ui/react';
+import type { FieldError } from 'react-hook-form';
+import { useForm } from 'react-hook-form'; // react-hook-form import 추가
+import { useLocation } from 'react-router-dom';
+
+export const Order = () => {
+  const location = useLocation();
+  const { productDetail, quantity, price } = location.state || {};
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm(); // useForm 훅을 사용하여 폼 상태와 메소드 초기화
```

Comment:
```tsx
interface FormValues {
  message: string;
  receiptNumber?: string;
  receiptRequested: boolean;
}
```

```suggestion
  useForm<FormValues>();
```

[handleSubimt](https://react-hook-form.com/docs/useform/handlesubmit) 공식 문서를 보면 어떤 key와 value가 가능한지 타입으로 명시해주고 있어요. 타입 안전하게 form value를 소비할 수 있돌고 타입을 전달해주는게 좋겠습니다.